src/daemon.cpp

```diff
@@ -19,239 +19,25 @@
-{
-    auto searchJob = static_cast<KIMAP::SearchJob*>(job);
-    if (searchJob->error()) {
-        qDebug() << searchJob->errorString();
-        return;
-    }
-    else {
-        qDebug() << "Search Job Done Successfully";
-    }
-
-    qDebug() << searchJob->results();
-    KIMAP::ImapSet set;
-    set.add(searchJob->results());
-    searchJob->kill();
-    KIMAP::FetchJob* fetchJob = new KIMAP::FetchJob(m_session);
-    fetchJob->setSequenceSet(set);
-    KIMAP::FetchJob::FetchScope scope = KIMAP::FetchJob::FetchScope();
-    scope.mode = KIMAP::FetchJob::FetchScope::Content;
-    fetchJob->setScope(scope);
-    connect(fetchJob, static_cast<CAST>(&KIMAP::FetchJob::messagesReceived), 
-            this, &Daemon::onFetchJobFinished);
-    fetchJob->start();
-}
-
-void Daemon::onFetchJobFinished(const QString& mailBox, const QMap< qint64, qint64 >& uids,
-                                  const QMap< qint64, KIMAP::MessagePtr >& messages)
-{
-    Q_UNUSED(mailBox)
-    Q_UNUSED(uids)
-    qDebug() << messages << "\n";
-
-    QMapIterator<qint64, KIMAP::MessagePtr> it(messages);
-    while (it.hasNext()) {
-        it.next();
-
-        qDebug() << it.value()->subject()->asUnicodeString();
-        emit fetchedEmail(it.value());
-    }
-
-    KSharedConfigPtr config = KSharedConfig::openConfig("kdenowrc");
-    KConfigGroup generalGroup(config, "General");
-    generalGroup.writeEntry("STATE", "Update");
-    generalGroup.config()->sync();
-
-    startIdle();
-}
-
-void Daemon::startIdle()
-{
-    KIMAP::IdleJob *idleJob = new KIMAP::IdleJob(m_session);
-    connect(idleJob, &KIMAP::IdleJob::mailBoxStats, this, &Daemon::onIdleChanged);
-    qDebug() << "IDLIing the Connection";
-    idleJob->start();
-}
-
-void Daemon::onIdleChanged(KIMAP::IdleJob* job, const QString& mailBox,
-                           int messageCount, int recentCount)
-{
-    Q_UNUSED(mailBox)
-    Q_UNUSED(recentCount)
-    qDebug() << "Update !";
-
-    KSharedConfigPtr config = KSharedConfig::openConfig("kdenowrc");
-    KConfigGroup generalGroup(config, "General");
-    qint64 currentMessageCount = generalGroup.readEntry("MESSAGE_COUNT", QString()).toULongLong();
-
-    if (messageCount < currentMessageCount) {
-        qDebug() << "User has deleted an email. Return";
-        generalGroup.writeEntry("MESSAGE_COUNT", QString::number(messageCount));
-        generalGroup.config()->sync();
-        return;
-    }
-    else {
-        qDebug() << "New email has arrived, so fetch it";
-        generalGroup.writeEntry("MESSAGE_COUNT", QString::number(messageCount));
-        generalGroup.config()->sync();
-        job->stop();
-        m_session->close();
-        emit signalUpdateProcess();
+void Daemon::setCredentials(QList < UserCredentials >& credentialsList) {
+    Processor* processor = SingletonFactory::instanceFor< Processor >();
+    foreach (UserCredentials credentials, credentialsList) {
+        EmailManager manager;
```

Comment:
This instance would be destroyed after returning from this function. Create a QList for EmailManager as a member variable and append dynamically allocated EmailManager to that list.